src/ray/raylet/node_manager.cc

```diff
@@ -447,6 +447,36 @@ void NodeManager::ProcessClientMessage(
       worker->MarkUnblocked();
     }
   } break;
+  case protocol::MessageType_WaitRequest: {
+    // Read the data.
+    auto message = flatbuffers::GetRoot<protocol::WaitRequest>(message_data);
+    auto object_id_strings = message->object_ids();
+    int num_requests = object_id_strings->size();
+    std::vector<ObjectID> object_ids;
+    for (int i = 0; i < num_requests; i++) {
+      ObjectID object_id = ObjectID::from_binary(object_id_strings->Get(i)->str());
+      object_ids.push_back(object_id);
+    }
+    int64_t wait_ms = message->timeout();
+    uint64_t num_required_objects = static_cast<uint64_t>(message->num_ready_objects());
+    bool wait_local = message->wait_local();
+
+    ray::Status status = object_manager_.Wait(
+        object_ids, wait_ms, num_required_objects, wait_local,
+        [this, client](int64_t time_taken, std::unordered_set<ObjectID> found,
+                       std::unordered_set<ObjectID> remaining) {
+          // Write the data.
+          std::vector<ObjectID> found_vec(found.begin(), found.end());
+          std::vector<ObjectID> remaining_vec(remaining.begin(), remaining.end());
+          flatbuffers::FlatBufferBuilder fbb;
+          flatbuffers::Offset<protocol::WaitReply> wait_reply = protocol::CreateWaitReply(
+              fbb, to_flatbuf(fbb, found_vec), to_flatbuf(fbb, remaining_vec));
+          fbb.Finish(wait_reply);
+          client->WriteMessage(protocol::MessageType_WaitReply, fbb.GetSize(),
```

Comment:
If the client dies in the middle of `client->WriteMessage`, are we handling that? Where will that error be handled?

Reply:
I added a check here; if the write fails an `IOError` status is returned.